### DISCUSSION TITLE
CodeReady workspace with Apache, PHP, MariaDB and Eclipse Theia

### DISCUSSION THREAD

**Discussion by Verhoeckx** [cockpit-project/cockpit#16511] [Q&A] [answered]
Hello Cockpit developers,

A few days ago I tried Red Hat CodeReady workspaces on OpenShift and now I'm wondering: is it possible to create such a setup on Cockpit?

In other words: I'm looking for a container that contains:

- Apache
- PHP (+ Composer)
- MariaDB
- Eclipse Theia (web IDE)

Does such a container exist and is it possible to run it on Cockpit with Podman?
&nbsp;
&nbsp;
On [Red Hat Catalog](https://catalog.redhat.com/software/containers/explore) I found the following two images:

Red Hat CodeReady Workspaces - PHP Stack
This image contains Apache and PHP, but not MariaDB and Eclipse Theia.

Red Hat CodeReady Workspaces - Theia container 
This image only contains Eclipse Theia (I guess). It doesn't contain Apache, PHP and MariaDB.
&nbsp;
&nbsp;
The main goal is to setup a complete PHP environment. If there doesn't exist a container for this I will setup a virtual machine with the LAMP stack and, if possible, also with Eclipse Theia.

**Comment by martinpitt** [comment]
I had a [quick](https://hub.docker.com/search?q=codeready&type=image) [look](https://hub.docker.com/search?q=theia&type=image) on dockerhub, and didn't find any recent image. However, it would be rather unusual to have so many things packed into a single container image, that's kind of besides their point -- you'd usually run e.g. a mariadb container image and some application container side by side, in a pod (with podman) or as two separate containers (with older docker). In principle you can run pretty much any application container with podman.

None of this is cockpit related, though.

**Comment by Verhoeckx** [comment]
Hello @martinpitt,

Thanks for your explanation!

My question is indeed not completely related to Cockpit but it would be great if it would be possible to create a complete PHP development environment with one click on Cockpit using `cockpit-podman`.

About containers: I have the impression they can be used either way: several containers working together, or one container with multiple applications (for example what Toolbox does).

It would be great if you had a pod that contained everything that's needed to setup a PHP development but I don't think something like this already exists (and Cockpit is able to run it)?

The only way at the moment is create a virtual machine that contains everything or you have to setup multiple containers manually?

**Comment by Verhoeckx** [comment, marked as answer]
Will/does `cockpit-podman` support Docker compose?

**Reply by jelly** [reply]
Not at the moment, there is an issue to support it https://github.com/cockpit-project/cockpit-podman/issues/247

**Reply by Verhoeckx** [reply]
Thanks, yes, that would be great! 
Actually, now I see the issue, I remember that I have read it a few months ago :thinking: !

**Comment by Verhoeckx** [comment]
I guess CodeReady workspaces uses something like Docker compose to run a pod on OpenShift?
Would be great if it would be possible to run this with cockpit-compose on Cockpit.
